Libraries/rocJPEG/rocjpeg_decode/CMakeLists.txt

```
@@ -0,0 +1,70 @@
+# MIT License
+#
+# Copyright (c) 2025 Advanced Micro Devices, Inc. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+set(example_name rocjpeg_decode)
+
+cmake_minimum_required(VERSION 3.21 FATAL_ERROR)
+project(${example_name} LANGUAGES CXX)
+
+if(GPU_RUNTIME STREQUAL "CUDA")
+    message(STATUS "rocJPEG examples do not support the CUDA runtime")
+    return()
+endif()
+
+enable_language(${GPU_RUNTIME})
+set(CMAKE_${GPU_RUNTIME}_STANDARD 17)
+set(CMAKE_${GPU_RUNTIME}_EXTENSIONS OFF)
+set(CMAKE_${GPU_RUNTIME}_STANDARD_REQUIRED ON)
+
+if(CMAKE_SYSTEM_NAME MATCHES "Windows")
+    message(STATUS "rocJPEG examples are only available on Linux")
+    return()
+else()
+    set(ROCM_ROOT
+        "/opt/rocm"
+        CACHE PATH
+        "Root directory of the ROCm installation"
+    )
+endif()
+
+list(APPEND CMAKE_PREFIX_PATH "${ROCM_ROOT}")
+
+find_package(rocjpeg REQUIRED)
+
+add_executable(${example_name} main.cpp)
+
+# Link to example library
+target_link_libraries(${example_name} PRIVATE rocjpeg::rocjpeg stdc++fs)
```

rocm-examples provides a lookup for the filesystem library (to guard against future compilers not shipping that anymore):

```suggestion
include("${CMAKE_CURRENT_LIST_DIR}/../../../Common/FindFilesystem.cmake")
target_link_libraries(${example_name} PRIVATE rocjpeg::rocjpeg ${CXX_FS_LIBRARY})
```

---

I don't see where `<filesystem>` is used in the code. Is it possible to not link against it at all?

---

`<filesystem>` is used in `Common/rocjpeg_utils.hpp` so that's fine, but would be nice to use `FindFilesystem.cmake`

---

Fixed